sources/csv_reader/__init__.py

```diff
@@ -0,0 +1,69 @@
+"""
+Source, which uses the `filesystem` source and DuckDB
+to extract CSV files data from the given locations.
+"""
+from typing import Iterable, List
+
+import duckdb
+import pendulum
+
+import dlt
+from dlt.common.typing import TDataItem, TAnyDateTime
+
+try:
+    from filesystem import filesystem
+except ImportError:
+    from sources.filesystem import filesystem
+
+from fsspec.implementations.local import LocalFileSystem
+
+from .helpers import add_columns
+
+
+@dlt.resource
+def read_location(files):
+    """A resource to extract data from the given CSV files.
+
+    Args:
+        files (List[FileItem]): A list of files to read.
+
+    Returns:
+        Iterable[TDataItem]: Data items, read from the given CSV files.
+    """
+    state = dlt.current.resource_state()
+    start_from = state.setdefault("last_modified", pendulum.datetime(1970, 1, 1))
```

Comment:
can start_date be parameterized or will we actually have datastores with files from 70s?